### PR TITLE
Support pyarrow 17

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -3,6 +3,11 @@ Version history / changelog
 
 From version 2.0.0, turbodbc adapts semantic versioning.
 
+Version 4.13.0
+--------------
+
+* Support ``pyarrow=16`
```
Comment (alippai) [inline CHANGELOG.rst:9]:
```suggestion
* Support ``pyarrow=17`
```

File: Earthfile [hunk 2]
```diff
@@ -213,6 +213,24 @@ test-python3.10-arrow15.x.x:
...
+        --build-arg ARROW_VERSION_RULE=">=16,<17" \
+        --build-arg NUMPY_VERSION_RULE=">=1.21.2" \
+        +test/result /result
+
+    SAVE ARTIFACT /result AS LOCAL result
+
+test-python3.10-arrow15.x.x:
+    ARG PYTHON_VERSION="3.10.2"
+    COPY --build-arg PYTHON_VERSION="$PYTHON_VERSION" \
+        --build-arg ARROW_VERSION_RULE=">=17,<18" \
```
Comment (alippai) [inline Earthfile:228]: `arrow15` vs `>=17`

File: Earthfile [hunk 3]
```diff
@@ -213,6 +213,24 @@ test-python3.10-arrow15.x.x:
...
+test-python3.10-arrow16.x.x:
+    ARG PYTHON_VERSION="3.10.2"
+    COPY --build-arg PYTHON_VERSION="$PYTHON_VERSION" \
+        --build-arg ARROW_VERSION_RULE=">=16,<17" \
+        --build-arg NUMPY_VERSION_RULE=">=1.21.2" \
+        +test/result /result
+
+    SAVE ARTIFACT /result AS LOCAL result
+
+test-python3.10-arrow15.x.x:
```
Comment (xhochy) [inline Earthfile:225]:
```suggestion
test-python3.10-arrow17.x.x:
```

**Comment by alippai** [comment]
Thanks @xhochy !

**Pull request merged**